x-pack/test_serverless/shared/lib/security/kibana_roles/kibana_roles.ts

```diff
@@ -48,6 +49,34 @@ export const getServerlessSecurityKibanaRoleDefinitions = (
         `Un-expected role [${roleName}] found in YAML file [${ROLES_YAML_FILE_PATH}]`
       );
     }
+    const mapApplicationToKibanaFeaturePrivileges = (
+      application: IApplication
+    ): FeaturesPrivileges => {
+      if (application.resources !== '*') {
+        throw new Error(
+          `YAML role definition parser does not currently support 'application.resource = ${application.resources}' for ${application.application} `
+        );
+      }
+
+      const features: FeaturesPrivileges = {};
+
+      application.privileges.forEach((value) => {
+        const [feature, permission] = value.split('.');
+        const featureKey = feature.split('_')[1];
+
+        if (!features[featureKey]) {
+          features[featureKey] = [];
+        }
+
+        if (permission) {
+          features[featureKey].push(permission);
+        }
+      });
+
+      return features;
+    };
+
+    const feature = mapApplicationToKibanaFeaturePrivileges(definition.applications[0]);
```

Comment:
is there any specific reason to remove the `reduce` here?

Reply:
Not really, I just liked forEach better. Is it ok with you?